crates/polars-io/src/cloud/object_store_setup.rs

```diff
@@ -40,6 +40,14 @@ fn url_and_creds_to_key(url: &Url, options: Option<&CloudOptions>) -> String {
     )
 }
 
+/// Simply construct an object_store `Path` struct from a string.
+pub fn object_path_from_string(path: String) -> object_store::path::Path {
+    // We transmute because they don't expose a way to just create it from a string
+    // without encoding or decoding it. If one day we can't use this transmute hack
```

Comment:
I was running around in circles getting either an extra encode or an extra decode until I realized this 🥲

Reply:
Auch.. Can we open a feature request upstream? If they ever add a field to the Path, we are UB.

Reply:
Added a check for the alignment. The size should already be handled by `transmute`.

Opened a feature request upstream here - https://github.com/apache/arrow-rs/issues/5993

Reply:
As stated on the linked issue, this should not be necessary `Path::parse` doesn't alter the provided string